### DISCUSSION TITLE
Add thor help command to view all subcommands simultaneously

### DISCUSSION THREAD

**Discussion by hlascelles** [erikhuda/thor#905] [Ideas]
I would like to be able to see all thor commands and subcommands in one go.

ie not:

```bash
❯ myapp --help
Commands:
  myapp foo            # Foo commands
  myapp qux            # Qux commands

❯ myapp foo --help
Commands:
  myapp foo baz        # Baz!
  myapp foo bar        # Bar!
```

I would like:

```bash
❯ myapp --tree-help
Commands:
  myapp foo            # Foo commands
  myapp foo baz        # Baz!
  myapp foo bar        # Bar!
  myapp qux            # Qux commands
  myapp qux pop        # Pop!
```

The switch could be one of:

```bash
myapp --tree-help
myapp  --help-tree
myapp --help --tree
myapp --tree
....
```

If you think this would be useful I could have a go... Or does something like it exist?

**Comment by hlascelles** [comment]
And/or to put it another way, how do I find a command where I know the subcommand is `baz`? (That happens to be my usecase).

But I would also like to print out a tree too!

**Comment by hlascelles** [comment]
@rafaelfranca @yahonda If I had a go at this would it be well received?

**Comment by rafaelfranca** [comment]
I think if it is a flag to the command, I see no problem with doing this.

**Comment by hlascelles** [comment]
@rafaelfranca I've had an initial go here: https://github.com/rails/thor/pull/906

It exposes it as a simple command like `app tree` and as `app --tree`. Let me know how you would like it improved!

<img width="1397" height="283" alt="image" src="https://github.com/user-attachments/assets/0ce786ab-6480-4d89-9933-9ae1612e06e6" />
